lib/Dialect/P4HIR/P4HIR_Ops.cpp

```diff
@@ -806,11 +806,21 @@ void P4HIR::VariableOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
 }
 
 LogicalResult P4HIR::VariableOp::canonicalize(P4HIR::VariableOp op, PatternRewriter &rewriter) {
+    auto users = op->getUsers();
+    auto firstNonAssignOp =
+        llvm::find_if(users, [](auto *user) { return !mlir::isa<P4HIR::AssignOp>(user); });
+
+    if (firstNonAssignOp == users.end()) {
+        // Completely remove variable if it is only written to.
```

Comment:
This is just an optimization here, right? The pass itself does not depend on this?